imageprocessing/src/main/java/com/example/imageprocessing/PreProcessing.java

```diff
@@ -0,0 +1,394 @@
+                      We called beta as changeBrightness*/
+                    int alpha = 1;
+
+                    /*Modify the values of all pixels with an alpha and beta value following
+                      the formula above.*/
+                    bright.convertTo(modifiedMat, matrixType, alpha, changeBrightness);
+
+                    //Verify if the image is good enough
+                    if(isBright(modifiedMat) == BrightnessValue.IMAGE_IS_OK){
+                        try{
+                            return IPUtils.conversionMatToBitmap(modifiedMat);
+                        } catch (ConversionFailedException error){
+                            Log.e(TAG, error.getErrorMessage());
+                            return image;
+                        }
+                    }
+                }
+                break;
+
+            case IMAGE_TOO_DARK:
+                Log.d(TAG, "Case==IMAGE_TOO_DARK");
+                //The variables are explained in the case above
+                //Lightens the colour's brightness until it's in an optimal value
+                for(double changeBrightness = 0; changeBrightness != maxBrightness;
+                    changeBrightness += step){
+
+                    //Converts an array to another data type with optional scaling. The variables's
+                    // values are explained in case 1
+                    int matrixType = -1;
+                    int alpha = 1;
+                    bright.convertTo(modifiedMat, matrixType, alpha, changeBrightness);
+
+                    //Verify if the image is good enough
+                    if(isBright(modifiedMat) == BrightnessValue.IMAGE_IS_OK){
+                        //If the conversion failed it returns the original image
+                        try{
+                            return IPUtils.conversionMatToBitmap(modifiedMat);
+                        } catch (ConversionFailedException error){
+                            Log.e(TAG, error.getErrorMessage());
+                            return image;
+                        }
+                    }
+                }
+                break;
+
+            case IMAGE_IS_OK: //Image is neither too bright nor too dark
+                Log.d(TAG, "Case==IMAGE_IS_OK");
+                return image;
+        }
+        return image;
+    }
+
+
+    /**
+     * @author Thomas Porro(g1), Oscar Garrido (g1), Giovanni Fasan(g1).
+     * See PreProcessingMethods.java
+     */
+    @Override
+    public boolean isBlurred(Bitmap image) {
+
+        //Total number of color
+        int maxLap = -16777216;
+
+        //Threshold above which the color is out of focus
+        final int threshold = -6118750;
+
+        //Converts the image into a matrix
+        Mat imageMat;
+        try {
+            imageMat = IPUtils.conversionBitmapToMat(image);
+        } catch (ConversionFailedException error){
+            Log.e(TAG, error.getErrorMessage());
+            return false;
```

Comment:
This return like the other one at line 357 it's not correct. Indeed if an error occurred while converting something into a matrix or a bitmap the method's result is false and this means "the given image isn't blurred" when this could not be true. If I where you, I've returned a different value or do something different to distinguish if something went wrong or if the image is really not blurred.

Reply:
Ok, then instead of boolean I'll do an enum to differentiate the situations.